### PR TITLE
Mr15 implement robot upgrading

### PR DESCRIPTION
**Pull request by Chris-TH**

### CODE REVIEW

**Comment by DennisWae** [comment]
Please don't put the code of other tickets into unrelated pull requests, as this makes them harder to review. Please revert your previoues two commits and create new branches for them and open new pull requests.

**Review by DennisWae — CHANGES_REQUESTED** [review]
Please revert your previous two commits

**Review by DennisWae — CHANGES_REQUESTED** [review]
Wait until MR8 ist merged into main and then merge main into this branch.

**Comment by DennisWae** [comment]
MR8 is now merged.

**Comment by Chris-TH** [comment]
Merged branch 'main' into MR15-Implement-Robot-Upgrading

File: src/main/kotlin/com/msd/robot/application/RobotApplicationService.kt [hunk 1]
```diff
@@ -120,6 +121,21 @@ class RobotApplicationService(
...
+     * @param robotId     the `Robot` which should be updated
+     * @param upgradeType The upgrade which should increase its level
+     * @param level       the level to which the upgrade should increase
+     * @throws RobotNotFoundException  if there is not `Robot` with the specified ID
+     * @throws UpgradeException        if there is an attempt to skip a level, downgrade or upgrade past the max level
+     */
+    fun upgrade(robotId: UUID, upgradeType: UpgradeType, level: Int) {
+        val robot = robotDomainService.getRobot(robotId)
+        robot.upgrade(upgradeType, level)
+    }
```
Comment (DennisWae) [inline src/main/kotlin/com/msd/robot/application/RobotApplicationService.kt:136]: You need to persist the Robot after modifying it.

**Review by DennisWae — CHANGES_REQUESTED** [review]

**Review by DennisWae — APPROVED** [review]

**Pull request merged**